### PR TITLE
sip: add sticky buffers to match headers v6

### PR DESCRIPTION
**Pull request by glongo**
Make sure these boxes are signed before submitting your Pull Request -- thank you.

- [x] I have read the contributing guide lines at
   https://docs.suricata.io/en/latest/devguide/contributing/contribution-process.html
- [x] I have signed the Open Information Security Foundation contribution agreement at
   https://suricata.io/about/contribution-agreement/ (note: this is only required once)
- [x] I have updated the user guide (in doc/userguide/) to reflect the changes made (if applicable)
- [x] I have created a ticket at
      https://redmine.openinfosecfoundation.org/projects/suricata/issues
      (if applicable)

Link to ticket:
https://redmine.openinfosecfoundation.org/issues/6374
https://redmine.openinfosecfoundation.org/issues/7204

Describe changes:
- Rebase
- Add test for multi-value header

SV_BRANCH=https://github.com/OISF/suricata-verify/pull/2005

### CODE REVIEW

**Comment by catenacyber** [comment]
I did not do a full review, but this looks good to me

**Comment by catenacyber** [comment]
@glongo for your info, https://github.com/OISF/suricata/pull/11699 will come to help transition the other sip keywords to rust (ones that need validate callbacks)

**Review by catenacyber — APPROVED** [review]
Thanks for the nice work.

CI : ✅
Code : good
Commits segmentation : ok for me (split in detailed steps)
Commit messages : ok, last doc commit can be linked to Ticket #6374 as well
Git ID set : looks fine for me
CLA : you already contributed :-p
Doc update : nice
Redmine ticket : ok (I did some updates on https://redmine.openinfosecfoundation.org/issues/7204 )
Rustfmt : cool
Tests : nice, thanks
Dependencies added: none

File: rust/src/sip/detect.rs [hunk 1]
```diff
@@ -80,88 +126,662 @@ pub unsafe extern "C" fn rs_sip_tx_get_protocol(
...
-            return 1;
+            return true;
         }
     }
+    *buffer = ptr::null();
+    *buffer_len = 0;
+    return false;
+}
 
+unsafe extern "C" fn sip_get_header_value(
```
Comment (catenacyber) [inline rust/src/sip/detect.rs:312]: nit but this does not need to be `extern "C"` as this is pure rust

It should return an `Option<&[u8]>`

The `unsafe` part should be handled by the callers why are already unsafe extern C

File: rust/src/sip/detect.rs [hunk 2]
```diff
@@ -80,88 +126,662 @@ pub unsafe extern "C" fn rs_sip_tx_get_protocol(
...
+    tx: *const c_void, i: u32, direction: u8, buffer: *mut *const u8, buffer_len: *mut u32,
+    strname: *const c_char,
```
Comment (catenacyber) [inline rust/src/sip/detect.rs:314]: `strname` should use a pure rust type instead of converting back and forth

Comment (glongo) [reply rust/src/sip/detect.rs:314]: As I reported in the previous PR (https://github.com/OISF/suricata/pull/11620#discussion_r1734995838):
```
The compiler complains about that:
extern fn uses type `str`, which is not FFI-safe

Same for string.
```

Comment (catenacyber) [reply rust/src/sip/detect.rs:314]: Yes, this function `sip_get_header_value should not be labelled `extern` :

You can squash this commit in 
https://github.com/OISF/suricata/commit/6b7098d2f7d83c88b2369d86a00a911466bdb572

**Review by victorjulien — APPROVED** [review]

**Comment by victorjulien** [comment]
Merged in #11809, thanks!

**Pull request closed without merging**